### PR TITLE
update PHP dependencies and remove communication feature

### PR DESCRIPTION
**Pull request by longrunningprocess**
## Description

This PR seeks to update our PHP deps.  While in the process of upgrading, it was noticed that our SMS feature was never rolled out and was keeping the update from continuing without further research.  It was decided that we could simply remove all the SMS functionality.

### Type of Change

- dependency update and residual unused features

### SMS feature before

image

### SMS feature after

image

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title & description of this PR which I will use as the squashed PR commit message
- [x] I have enabled auto-merge (optional)
- [x]  rebase off `develop` after #1610 is merged

## How to test

- verify automated tests passed
- general smoke test of app
- create an account to ensure emails are unaffected
- ensure communication preference section is gone from "My Profile"

## qa.languageforge.org testing

Testers should add his/her findings to end of the PR in a comment and include screenshots, files, etc that are beneficial.

### CODE REVIEW

**Comment by longrunningprocess** [comment]
```
# composer update
Loading composer repositories with package information
Info from https://repo.packagist.org: #StandWithUkraine
Updating dependencies
Lock file operations: 0 installs, 40 updates, 6 removals
  - Removing phpdocumentor/reflection-common (1.0.1)
  - Removing phpdocumentor/reflection-docblock (4.3.1)
  - Removing phpdocumentor/type-resolver (0.4.0)
  - Removing phpspec/prophecy (1.8.1)
  - Removing twilio/sdk (dev-master 45a00de)
  - Removing webmozart/assert (1.4.0)
  - Upgrading doctrine/instantiator (1.2.0 => 1.4.1)
  - Upgrading firebase/php-jwt (v5.2.1 => v5.5.1)
  - Upgrading guzzlehttp/promises (1.5.1 => 1.5.2)
  - Upgrading league/oauth2-client (2.4.1 => 2.6.1)
  - Upgrading league/oauth2-facebook (2.0.1 => 2.2.0)
  - Upgrading myclabs/deep-copy (1.9.1 => 1.11.0)
  - Upgrading paragonie/random_compat (v9.99.99 => v9.99.100)
  - Upgrading phar-io/manifest (1.0.3 => 2.0.3)
  - Upgrading phar-io/version (2.0.1 => 3.2.1)
  - Upgrading phpunit/php-code-coverage (7.0.7 => 7.0.15)
  - Upgrading phpunit/php-file-iterator (2.0.2 => 2.0.5)
  - Upgrading phpunit/php-timer (2.1.2 => 2.1.3)
  - Upgrading phpunit/php-token-stream (3.1.0 => 4.0.4)
  - Upgrading phpunit/phpunit (8.3.3 => 8.5.31)
  - Upgrading psr/log (1.1.1 => 1.1.4)
  - Upgrading ramsey/uuid (3.8.0 => 3.9.6)
  - Upgrading sebastian/code-unit-reverse-lookup (1.0.1 => 1.0.2)
  - Upgrading sebastian/comparator (3.0.2 => 3.0.5)
  - Upgrading sebastian/diff (3.0.2 => 3.0.3)
  - Upgrading sebastian/environment (4.2.2 => 4.2.4)
  - Upgrading sebastian/exporter (3.1.0 => 3.1.5)
  - Upgrading sebastian/global-state (3.0.0 => 3.0.2)
  - Upgrading sebastian/object-enumerator (3.0.3 => 3.0.4)
  - Upgrading sebastian/object-reflector (1.1.1 => 1.1.2)
  - Upgrading sebastian/recursion-context (3.0.0 => 3.0.1)
  - Upgrading sebastian/resource-operations (2.0.1 => 2.0.2)
  - Upgrading sebastian/type (1.1.3 => 1.1.4)
  - Upgrading squizlabs/php_codesniffer (3.4.2 => 3.7.1)
  - Upgrading symfony/debug (v3.4.30 => v3.4.47)
  - Upgrading symfony/polyfill-ctype (v1.26.0 => v1.27.0)
  - Upgrading symfony/polyfill-intl-idn (v1.26.0 => v1.27.0)
  - Upgrading symfony/polyfill-intl-normalizer (v1.26.0 => v1.27.0)
  - Upgrading symfony/polyfill-mbstring (v1.26.0 => v1.27.0)
  - Upgrading symfony/polyfill-php56 (v1.12.0 => v1.20.0)
  - Upgrading symfony/polyfill-php72 (v1.26.0 => v1.27.0)
  - Upgrading symfony/polyfill-util (v1.12.0 => v1.27.0)
  - Upgrading symfony/security (v2.8.50 => v2.8.52)
  - Upgrading symfony/security-acl (v3.0.2 => v3.0.4)
  - Upgrading symfony/twig-bridge (v2.8.50 => v2.8.52)
  - Upgrading theseer/tokenizer (1.1.3 => 1.2.1)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 0 installs, 40 updates, 6 removals
  - Downloading firebase/php-jwt (v5.5.1)
  - Downloading symfony/polyfill-php72 (v1.27.0)
  - Downloading symfony/polyfill-intl-normalizer (v1.27.0)
  - Downloading symfony/polyfill-intl-idn (v1.27.0)
  - Downloading guzzlehttp/promises (1.5.2)
  - Downloading paragonie/random_compat (v9.99.100)
  - Downloading league/oauth2-client (2.6.1)
  - Downloading league/oauth2-facebook (2.2.0)
  - Downloading phpunit/php-token-stream (4.0.4)
  - Downloading sebastian/type (1.1.4)
  - Downloading sebastian/resource-operations (2.0.2)
  - Downloading sebastian/recursion-context (3.0.1)
  - Downloading sebastian/object-reflector (1.1.2)
  - Downloading sebastian/object-enumerator (3.0.4)
  - Downloading sebastian/global-state (3.0.2)
  - Downloading sebastian/exporter (3.1.5)
  - Downloading sebastian/environment (4.2.4)
  - Downloading sebastian/diff (3.0.3)
  - Downloading sebastian/comparator (3.0.5)
  - Downloading phpunit/php-timer (2.1.3)
  - Downloading phpunit/php-file-iterator (2.0.5)
  - Downloading theseer/tokenizer (1.2.1)
  - Downloading sebastian/code-unit-reverse-lookup (1.0.2)
  - Downloading phpunit/php-code-coverage (7.0.15)
  - Downloading phar-io/version (3.2.1)
  - Downloading phar-io/manifest (2.0.3)
  - Downloading myclabs/deep-copy (1.11.0)
  - Downloading symfony/polyfill-mbstring (v1.27.0)
  - Downloading doctrine/instantiator (1.4.1)
  - Downloading phpunit/phpunit (8.5.31)
  - Downloading symfony/polyfill-ctype (v1.27.0)
  - Downloading ramsey/uuid (3.9.6)
  - Downloading psr/log (1.1.4)
  - Downloading symfony/debug (v3.4.47)
  - Downloading squizlabs/php_codesniffer (3.7.1)
  - Downloading symfony/polyfill-util (v1.27.0)
  - Downloading symfony/security-acl (v3.0.4)
  - Downloading symfony/security (v2.8.52)
  - Downloading symfony/twig-bridge (v2.8.52)
  - Removing webmozart/assert (1.4.0)
  - Removing twilio/sdk (dev-master 45a00de)
  - Removing phpspec/prophecy (1.8.1)
  - Removing phpdocumentor/type-resolver (0.4.0)
  - Removing phpdocumentor/reflection-docblock (4.3.1)
  - Removing phpdocumentor/reflection-common (1.0.1)
  - Upgrading firebase/php-jwt (v5.2.1 => v5.5.1): Extracting archive
  - Upgrading symfony/polyfill-php72 (v1.26.0 => v1.27.0): Extracting archive
  - Upgrading symfony/polyfill-intl-normalizer (v1.26.0 => v1.27.0): Extracting archive
  - Upgrading symfony/polyfill-intl-idn (v1.26.0 => v1.27.0): Extracting archive
  - Upgrading guzzlehttp/promises (1.5.1 => 1.5.2): Extracting archive
  - Upgrading paragonie/random_compat (v9.99.99 => v9.99.100): Extracting archive
  - Upgrading league/oauth2-client (2.4.1 => 2.6.1): Extracting archive
  - Upgrading league/oauth2-facebook (2.0.1 => 2.2.0): Extracting archive
  - Upgrading phpunit/php-token-stream (3.1.0 => 4.0.4): Extracting archive
  - Upgrading sebastian/type (1.1.3 => 1.1.4): Extracting archive
  - Upgrading sebastian/resource-operations (2.0.1 => 2.0.2): Extracting archive
  - Upgrading sebastian/recursion-context (3.0.0 => 3.0.1): Extracting archive
  - Upgrading sebastian/object-reflector (1.1.1 => 1.1.2): Extracting archive
  - Upgrading sebastian/object-enumerator (3.0.3 => 3.0.4): Extracting archive
  - Upgrading sebastian/global-state (3.0.0 => 3.0.2): Extracting archive
  - Upgrading sebastian/exporter (3.1.0 => 3.1.5): Extracting archive
  - Upgrading sebastian/environment (4.2.2 => 4.2.4): Extracting archive
  - Upgrading sebastian/diff (3.0.2 => 3.0.3): Extracting archive
  - Upgrading sebastian/comparator (3.0.2 => 3.0.5): Extracting archive
  - Upgrading phpunit/php-timer (2.1.2 => 2.1.3): Extracting archive
  - Upgrading phpunit/php-file-iterator (2.0.2 => 2.0.5): Extracting archive
  - Upgrading theseer/tokenizer (1.1.3 => 1.2.1): Extracting archive
  - Upgrading sebastian/code-unit-reverse-lookup (1.0.1 => 1.0.2): Extracting archive
  - Upgrading phpunit/php-code-coverage (7.0.7 => 7.0.15): Extracting archive
  - Upgrading phar-io/version (2.0.1 => 3.2.1): Extracting archive
  - Upgrading phar-io/manifest (1.0.3 => 2.0.3): Extracting archive
  - Upgrading myclabs/deep-copy (1.9.1 => 1.11.0): Extracting archive
  - Upgrading symfony/polyfill-mbstring (v1.26.0 => v1.27.0): Extracting archive
  - Upgrading doctrine/instantiator (1.2.0 => 1.4.1): Extracting archive
  - Upgrading phpunit/phpunit (8.3.3 => 8.5.31): Extracting archive
  - Upgrading symfony/polyfill-ctype (v1.26.0 => v1.27.0): Extracting archive
  - Upgrading ramsey/uuid (3.8.0 => 3.9.6): Extracting archive
  - Upgrading psr/log (1.1.1 => 1.1.4): Extracting archive
  - Upgrading symfony/debug (v3.4.30 => v3.4.47): Extracting archive
  - Upgrading squizlabs/php_codesniffer (3.4.2 => 3.7.1): Extracting archive
  - Removing symfony/polyfill-php56 (v1.12.0)
  - Upgrading symfony/polyfill-util (v1.12.0 => v1.27.0): Extracting archive
  - Upgrading symfony/security-acl (v3.0.2 => v3.0.4): Extracting archive
  - Upgrading symfony/security (v2.8.50 => v2.8.52): Extracting archive
  - Upgrading symfony/twig-bridge (v2.8.50 => v2.8.52): Extracting archive
 29/40 [====================>-------]  72%  - Installing symfony/polyfill-php56 (v1.20.0)
Package silex/silex is abandoned, you should avoid using it. Use symfony/flex instead.
Package swiftmailer/swiftmailer is abandoned, you should avoid using it. Use symfony/mailer instead.
Package symfony/debug is abandoned, you should avoid using it. Use symfony/error-handler instead.
Package twig/extensions is abandoned, you should avoid using it. No replacement was suggested.
Package phpunit/php-token-stream is abandoned, you should avoid using it. No replacement was suggested.
Generating autoload files
36 packages you are using are looking for funding.
```

**Comment by megahirt** [comment]
@longrunningprocess I would suggest that the name of this PR actually be "remove SMS feature and update PHP dependencies"

**Comment by megahirt** [comment]
This PR also needs to remove the SMS toggle from the UI.  We really just need to remove the Ui component for communication preference all together.

See:
image

File: src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts [hunk 1]
```diff
@@ -100,10 +100,6 @@ export class LexiconProjectService {
     return this.api.call('lex_project_update', [settings], callback);
   }
 
-  updateSettings(smsSettings: any, emailSettings: any, callback?: JsonRpcCallback) {
-    return this.api.call('project_updateSettings', [smsSettings, emailSettings], callback);
-  }
-
```
Comment (megahirt) [inline src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts:106]: removing this api call looks suspicious to me, mostly because the name of the api doesn't mention anything about sms or email, just the parameters.

There is a big rabbit trail here - if you remove the client side api calling code, should you not also remove the api code in the server?  It goes on and on I think.

File: src/angular-app/bellows/apps/userprofile/user-profile-app.component.html [hunk 2]
```diff
@@ -68,34 +68,6 @@
                                 </div>
                             </div>
                         </div>
-                        <div class="form-group row">
-                            <div class="card bg-light col-md-9 col-lg-7">
-                                <div class="card-body">
-                                    <div class="form-group row">
-                                        <label class="col-form-label col-md-4 text-md-right" for="mobile_phone">Mobile Phone Number</label>
-                                        <div class="col-md-8">
-                                            <input class="form-control" type="tel" data-pui-intl-tel-input id="mobile_phone" placeholder="###-####" data-ng-model="$ctrl.user.mobile_phone">
```
Comment (megahirt) [inline src/angular-app/bellows/apps/userprofile/user-profile-app.component.html:77]: removing the intl-tel-input will be a big win!  We can do a separate PR to remove this dependency and also JQuery, on which this old piece depends.

File: src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts (hunk 1 above)
Comment (megahirt) [reply src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts:106]: ok, I've done more research here and the updateSettings and readSettings are dead wood - not used anywhere but the end points still exist.

Please delete both of these methods in the AngularJS client (you've done one here) and delete traces of this back into PHP and PHP tests.  Thanks!

File: src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts [hunk 3]
```diff
@@ -100,10 +100,6 @@ export class LexiconProjectService {
...
   readSettings(callback?: JsonRpcCallback) {
```
Comment (megahirt) [inline src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts:103]: delete this method and the endpoint associated with it.

**Review by megahirt — CHANGES_REQUESTED** [review]
This is great work!  Looking forward to having this in.

Requesting changes to completely remove updateSettings and readSettings, as well as remove intl-tel-input from package.json

File: src/Api/Model/Shared/UserModel.php [hunk 4]
```diff
@@ -14,7 +14,6 @@
 
 class UserModel extends MapperModel
 {
-    const COMMUNICATE_VIA_SMS = "sms";
     const COMMUNICATE_VIA_EMAIL = "email";
     const COMMUNICATE_VIA_BOTH = "both";
```
Comment (myieye) [inline src/Api/Model/Shared/UserModel.php:18]: This should be gone too.

```suggestion
```

File: src/Api/Model/Shared/UserModel.php [hunk 5]
```diff
@@ -161,7 +160,7 @@ public function __construct($id = "")
     /** @var string */
     public $mobile_phone;
 
-    /** @var string - possible values are "email", "sms" or "both" */
+    /** @var string - possible values are "email" or "both" */
     public $communicate_via;
```
Comment (myieye) [inline src/Api/Model/Shared/UserModel.php:164]: In fact this whole property should be gone.

```suggestion
```

File: test/e2e/pages/user-profile.page.ts [hunk 6]
```diff
@@ -13,12 +13,6 @@ export class UserProfilePage extends BasePage {
     usernameField: this.page.getByLabel('Username'),
     colorField: this.page.locator('select:has-text("Select a Color...")'),
     animalField: this.page.locator('select:has-text("Choose an animal...")'),
-    phoneField: this.page.getByLabel('Mobile Phone Number'),
```
Comment (myieye) [inline test/e2e/pages/user-profile.page.ts:16]: There's nothing more satisfying than deleting fresh code 😆

File: src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts (hunk 3 above)
Comment (longrunningprocess) [reply src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts]: done in [c34a6b6](https://github.com/sillsdev/web-languageforge/pull/1611/commits/c34a6b62da664b8d2bc08e7a13096984be0b2ea1)

File: src/Api/Model/Shared/UserModel.php (hunk 4 above)
Comment (longrunningprocess) [reply src/Api/Model/Shared/UserModel.php]: done in [5b1e679](https://github.com/sillsdev/web-languageforge/pull/1611/commits/5b1e679ade7f12fa97f2e8df43021736665bebbe)

File: src/Api/Model/Shared/UserModel.php (hunk 5 above)
Comment (longrunningprocess) [reply src/Api/Model/Shared/UserModel.php]: yep!  [536d630](https://github.com/sillsdev/web-languageforge/pull/1611/commits/536d630a7185e7c219448dd7853b224a325392a5)

File: src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts (hunk 1 above)
Comment (longrunningprocess) [reply src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts:106]: done in [c34a6b6](https://github.com/sillsdev/web-languageforge/pull/1611/commits/c34a6b62da664b8d2bc08e7a13096984be0b2ea1)

File: src/angular-app/bellows/apps/userprofile/user-profile-app.component.html (hunk 2 above)
Comment (longrunningprocess) [reply src/angular-app/bellows/apps/userprofile/user-profile-app.component.html:77]: went ahead and took care of this while getting rid of the intl-tel-input, see [7bdbe26](https://github.com/sillsdev/web-languageforge/pull/1611/commits/7bdbe2635f75a8982f61f849f6a45d22bd41426e)

File: src/Api/Library/Shared/Communicate/Communicate.php [hunk 7]
```diff
@@ -47,47 +46,6 @@ public static function communicateToUsers(
...
 
-    /**
-     * @param UserModel $user
-     * @param ProjectSettingsModel $project
-     * @param string $subject
-     * @param string $emailTemplate
-     * @param string $htmlEmailTemplate
-     * @param DeliveryInterface $delivery
-     */
-    public static function communicateToUser(
```
Comment (megahirt) [inline src/Api/Library/Shared/Communicate/Communicate.php:70]: We still need this function as this is how the system sends emails.  The function should be left alone except for checks about the COMMUNICATE_VIA property.

File: src/angular-app/bellows/apps/public/signup/signup-app.component.html [hunk 8]
```diff
@@ -42,7 +42,6 @@ <h1>Create an account</h1>
                     </div>
                 </div>
                 <div>
-                    <script src="/dist/zxcvbn/zxcvbn.js"></script>
```
Comment (megahirt) [inline src/angular-app/bellows/apps/public/signup/signup-app.component.html:45]: Just want to verify that the password strength feature that zxcvbn provides still works properly after removing this.

File: src/angular-app/languageforge/lexicon/editor/editor.component.ts [hunk 9]
```diff
@@ -1277,15 +1277,15 @@ export class LexiconEditorController implements angular.IController {
     // asynchronously upon data retrieval...I'm not exactly sure what's needed here, but this is a poor-man's start.
     const entryDivId = '#entryId_' + id;
     const interval = this.$interval(() => {
-      if ($(entryDivId)[0]) {
```
Comment (megahirt) [inline src/angular-app/languageforge/lexicon/editor/editor.component.ts:1280]: nice!

File: src/app_dependencies.json [hunk 10]
```diff
@@ -1,5 +1,5 @@
 {
   "font-awesome": { "path": "dist/font-awesome/css", "cssFile": "font-awesome" },
-  "intl-tel-input": { "path": "dist/intl-tel-input", "jsFile": "js/intlTelInput", "cssFile": "css/intlTelInput" },
-  "ng-table": { "path": "dist/ng-table", "cssFile": "ng-table.min" }
```
Comment (megahirt) [inline src/app_dependencies.json:4]: this is great - we are slowly removing these.

File: test/php/library/shared/communicate/CommunicateTest.php [hunk 11]
```diff
@@ -38,101 +37,6 @@ public static function setUpBeforeClass(): void
         self::$environ = new MongoTestEnvironment();
     }
 
-    public function testCommunicateToUser_NoFromAddress_Exception()
```
Comment (megahirt) [inline test/php/library/shared/communicate/CommunicateTest.php:46]: we need to keep the tests since we still need the CommunicateToUser method for sending emails to users.  The tests just need to be fixed up after removing communicate_via

File: docker/next-app/Dockerfile.next-app [hunk 12]
```diff
@@ -19,4 +19,10 @@ COPY --from=builder /app/package.json /app/
 
 EXPOSE 3000
 
+# container would not receive SIGTERM from docker when shutting down so Docker would force kill
+# the container after 10s.  This will make shutdown faster because the SIGTERM will be handled appropriately.
+# https://maximorlov.com/process-signals-inside-docker-containers/
+RUN apk add --no-cache tini
+ENTRYPOINT ["/sbin/tini", "--"]
```
Comment (megahirt) [inline docker/next-app/Dockerfile.next-app:26]: thank you!  The next-app container was always the last to finish in `docker compose down`

**Review by megahirt — CHANGES_REQUESTED** [review]
Some great additions here!  Went a bit too far in removing our CommunicateToUser method which sends our emails.  Put that method and its tests back and this should be good to go.

File: src/Api/Library/Shared/Communicate/Communicate.php (hunk 7 above)
Comment (longrunningprocess) [reply src/Api/Library/Shared/Communicate/Communicate.php:70]: are you certain?  I don't think I could find the traceback for that, the emails related to registering and such are through different methods.

File: src/angular-app/bellows/apps/public/signup/signup-app.component.html (hunk 8 above)
Comment (longrunningprocess) [reply src/angular-app/bellows/apps/public/signup/signup-app.component.html:45]: I did test this locally and it still worked.

File: test/php/library/shared/communicate/CommunicateTest.php (hunk 11 above)
Comment (longrunningprocess) [reply test/php/library/shared/communicate/CommunicateTest.php:46]: I'd like to confirm that for sure if you don't mind, I really think these were related only to project update type comms.

File: src/Api/Library/Shared/Communicate/Communicate.php (hunk 7 above)
Comment (megahirt) [reply src/Api/Library/Shared/Communicate/Communicate.php:70]: 🤦 🥚 I stand corrected!  The communicateToUser() isn't used anymore as far as I can see.  The other method, `sendTemplateEmail()`, wraps `CommunicateHelper::deliverEmail()` which does the actual sending.  So this is all good.  Thanks for your patience as I get my head around all this dead wood code.  Glad we are slimming things down.

File: src/angular-app/bellows/apps/public/signup/signup-app.component.html (hunk 8 above)
Comment (megahirt) [reply src/angular-app/bellows/apps/public/signup/signup-app.component.html:45]: great thanks

File: test/php/library/shared/communicate/CommunicateTest.php (hunk 11 above)
Comment (megahirt) [reply test/php/library/shared/communicate/CommunicateTest.php:46]: You are right - we don't need CommunicateToUser.  Thanks for helping me see that.

File: src/Api/Model/Shared/UserModel.php (hunk 5 above)
Comment (megahirt) [reply src/Api/Model/Shared/UserModel.php]: thanks!

File: src/angular-app/bellows/apps/userprofile/user-profile-app.component.html (hunk 2 above)
Comment (megahirt) [reply src/angular-app/bellows/apps/userprofile/user-profile-app.component.html:77]: great thanks!

File: src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts (hunk 1 above)
Comment (megahirt) [reply src/angular-app/languageforge/lexicon/core/lexicon-project.service.ts:106]: great!

**Review by megahirt — APPROVED** [review]
In my previous review, I thought that CommunicateToUser() which this PR removes was still needed.  I was wrong, and that method isn't used at all, so this PR is good to go!  LGTM.

**Comment by laineyhm** [comment]
**Testing**
✅ Created an account
✅ No communication preferences in "My Profile"
✅ Smoke test
Looks good!

**Comment by longrunningprocess** [comment]
and you got the verification and intro emails @laineyhm ?

**Comment by laineyhm** [comment]
> and you got the verification and intro emails @laineyhm ?

Yes, thanks for confirming. Here's a welcome email from a new account created.
Screenshot (6)

**Comment by myieye** [comment]
**🧪 Testing**
- Created an account ✅
- Verified profile settings ✅
- Clicked around 💥 😉 => opened #1632 for a tiny bug I found.

**Comment by myieye** [comment]
Moved to waiting for PO signoff.
